unskript-ctl/unskript_ctl_config_parser.py

```diff
@@ -0,0 +1,229 @@
+        self.config_file = config_file
+        # Dictionary of jobs, with job name being the key.
+        self.jobs = {}
+
+    def parse_config_yaml(self) -> dict:
+        """parse_config_yaml: This function parses the config yaml file and converts the
+        content as a python dictionary and returns back to the caller.
+        """
+        retval = {}
+
+        if os.path.exists(self.config_file) == False:
+            print(f"WARNING: {self.config_file} Not found!")
+            return retval
+
+
+        # We use EnvYAML to parse the hook file and give us the
+        # dictionary representation of the YAML file
+        retval = EnvYAML(self.config_file, strict=False)
+
+        if not retval:
+            print(f"WARNING: config file {self.config_file} content seems to be empty!")
+            return retval
+
+        self.parsed_config = retval
+
+    def configure_credential(self):
+        """configure_credential: This function is used to parse through the creds_dict and
+        call the add_creds.sh method to populate the respective credential json
+        """
+        print('Processing credential section')
+        creds_dict = self.parsed_config.get('credential')
+        if creds_dict is None:
+            print(f"ERROR: Nothing to configure credential with, found empty creds data")
+            return
+
+        for cred_type in creds_dict.keys():
+            cred_list = creds_dict.get(cred_type)
+            for cred in cred_list:
+                if cred.get('enable') == False:
+                    continue
+                creds_cmd = ['/usr/local/bin/add_creds.sh', '-c', cred_type]
+                for cred_key in cred:
+                    # Skip name and enable keys
+                    if cred_key in ['name', 'enable']:
+                        continue
+                    # Certain arguments dont need extra value part like -no-verify-certs
+                    if cred_key in CREDENTIAL_CONFIG_SKIP_VALUE_FOR_ARGUMENTS:
+                        creds_cmd.extend(['--'+cred_key])
+                    else:
+                        creds_cmd.extend(['--'+cred_key, cred.get(cred_key)])
+                if creds_cmd:
+                    print_cmd = ' '.join(creds_cmd)
+                    print(f"Programming credential, command {print_cmd}")
+                    self.run_command(creds_cmd)
+                    print(f"Successfully programmed {cred_type}")
+                else:
+                    print(f"ERROR: No Credential {cred_type} was programmed")
+
+
+    def configure_schedule(self):
+        """configure_schedule: configures the schedule settings
+        """
+        print('Processing scheduler section')
+        config = self.parsed_config.get('scheduler')
+        if config is None:
+            print(f"No scheduler configuration found")
+            return
+
+        #unskript_crontab_file = "/unskript/etc/unskript_crontab.tab"
+        unskript_crontab_file = "./unskript_crontab.tab"
```

Comment:
It may be a good idea to make it at a well known path, like /tmp or something rather than `./`